app/js/RequestLogger.js

```diff
@@ -0,0 +1,83 @@
+const logger = require('logger-sharelatex')
+const metrics = require('metrics-sharelatex')
+
+module.exports = {
+  errorHandler,
+  middleware
+}
+
+function errorHandler(err, req, res, next) {
+  req._logInfo.set('error', err)
+  res
+    .send(err.message)
+    .status(500)
+    .end()
+}
+
+function middleware(req, res, next) {
+  const startTime = new Date()
+
+  // methods to allow the setting of additional information to be logged for the request
+  req._logInfo = {}
+  req._logMessage = 'http request'
+  req.addLogField = function(field, value) {
+    req._logInfo[field] = value
```

Comment:
This is a mega-nitpick, but I think I would probably just use `req.addLogFields({ foo })` (i.e. object shorthand for `{ foo: foo }` over the separate key/value arguments, therefore I'd argue for removing this. YMMV though.

Reply:
Yes, good point. Two ways of doing the same thing. Happy to remove it.